src/govuk/components/notification-banner/notification-banner.yaml

```diff
@@ -0,0 +1,177 @@
+params:
+- name: text
+  type: string
+  required: true
+  description: If `html` is set, this is not required. Text to use within the notification banner. If `html` is provided, the `text` argument will be ignored.
+- name: html
+  type: string
+  required: true
+  description: If `text` is set, this is not required. HTML to use within the notification banner. If `html` is provided, the `text` argument will be ignored.
+- name: title
+  type: string
+  required: false
+  description: Title text to use within the notification banner. Defaults to 'Important' ('Success' for success type and 'Error' for error type). If `titleHtml` is supplied, the `title` argument will be ignored.
+- name: titleHtml
+  type: string
+  required: false
+  description: Title HTML to use within the notification banner. If `titleHtml` is provided, the `title` argument will be ignored.
+- name: titleHeadingLevel
+  type: string
+  required: false
+  description: Heading level, from 1 to 6. Default is `2`.
+- name: type
+  type: string
+  required: false
+  description: If `type` is set to `success` or `error`, the notification banner sets `role` to `alert` and `tabindex` to `-1`, and JavaScript moves the keyboard focus moves to the notification banner when the page loads. If `type` is not set, the notification banner defaults to setting `role` to `region` and using `aria-labelledby` to provide information for users of assistive technologies.
```

Comment:
The description states what happens when you set type. Do we need to specify what type actually is, though?

Is this option to denote whether the notification banner is purely informational or else an alert box which the screen reader picks up on immediately?

Typo to remove: second "moves" in "Javascript moves the keyboard focus moves to the notification banner..."

Could personalise and make active the second last and last sentences. So, "If you set type to success..." and "If you do not set type..."

I'd also suggest changing "using aria-labelledby" to "uses aria-labelledby." This would be to parallel the active voice used in the previous clause ("...the notification banner defaults...").